kythe/typescript/testdata/for.ts

```diff
@@ -0,0 +1,8 @@
+export {};
+
+//- @x defines/binding X
+//- X.node/kind variable
+for (const x of [1, 2]) {
+  //- @x ref X
+  x;
+}
```

Comment:
Can you add some more examples, like standard for loops?

Reply:
Done.  Also it revealed some bugs!  So I fixed those too.